src/plugins_types.c

```diff
@@ -1335,29 +1398,40 @@ ly_type_store_empty(const struct ly_ctx *ctx, const struct lysc_type *type, cons
         const struct lysc_node *UNUSED(ctx_node), struct lyd_value *storage, struct lys_glob_unres *UNUSED(unres),
         struct ly_err_item **err)
 {
+    LY_ERR ret = LY_SUCCESS;
+
     /* check hints */
-    LY_CHECK_RET(type_check_hints(hints, value, value_len, type->basetype, NULL, err));
+    ret = type_check_hints(hints, value, value_len, type->basetype, NULL, err);
+    LY_CHECK_GOTO(ret != LY_SUCCESS, clean_up);
 
     if (value_len) {
         char *errmsg;
         if (asprintf(&errmsg, "Invalid empty value \"%.*s\".", (int)value_len, value) == -1) {
             *err = ly_err_new(LY_LLERR, LY_EMEM, 0, LY_EMEM_MSG, NULL, NULL);
-            return LY_EMEM;
+            ret = LY_EMEM;
         } else {
             *err = ly_err_new(LY_LLERR, LY_EVALID, LYVE_DATA, errmsg, NULL, NULL);
-            return LY_EVALID;
+            ret = LY_EVALID;
         }
+        goto clean_up;
     }
 
     if (options & LY_TYPE_STORE_DYNAMIC) {
-        LY_CHECK_RET(lydict_insert_zc(ctx, (char *)value, &storage->canonical));
+        ret = lydict_insert_zc(ctx, (char *)value, &storage->canonical);
+        options &= ~LY_TYPE_STORE_DYNAMIC;
+        /* LY_CHECK_GOTO(ret != LY_SUCCESS, clean_up); */
```

Comment:
seem like it is not finished - move the rest of the `storage` filling before storing the value or uncomment those `LY_CHECK_GOTO`s